adding new tables

PaulMicu opened the pull request:
As described in #98 I added 2 new tables for the asset_type and the trained_model. The asset_type table also holds boundaries for the assessment calculation

this script needs to be run after the v3 script you can find in the corresponding Zip in the script folder

here is an updated visual interpretation of the database to help review the pull request
![V4 01](https://user-images.githubusercontent.com/35611221/98756299-4552fd80-2398-11eb-8602-d6108b599043.png)

PaulMicu committed:
adding new tables

as described in #98 i added 2 new tables for the asset_type and the trained_model. The asset_type table also holds boundaries for the assesment calculation

PaulMicu committed:
Update modelV4.mwb

corrected model, wrong one uploaded previously

JayChouteau commented on src/main/resources/script/cbms_scriptsV4/V4.01.sql:
```diff
@@ -0,0 +1,73 @@
+*/
+UPDATE asset SET asset_type_id = 1 WHERE asset_id>0;
+
+select * from asset
```
The statement should end with an end of the line ";" symbol

JayChouteau commented on src/main/resources/script/cbms_scriptsV4/V4.01.sql:
```diff
@@ -0,0 +1,73 @@
+INSERT INTO asset_type(name, boundary_replace, boundary_repair) VALUES("Engine","99.8","55.5");
+
+/* 
+Adding the asset type id "1" to all current asset
```
Maybe leaving a TODO for adding system types desired names would be a good idea to remember to come back to it. This coding action isn't part of the scope of this PR and needs to come from the PO, but a TODO would help us remember once we get that information.

PaulMicu replied:
this will be part of issue #48, for now it needs something in the table for the assets we already have, we can change the name of the current type of assets at anytime to whatever the PO wants

PaulMicu replied:
This will be addressed in issue #48. The name of the asset type can be changed at any time

JayChouteau commented on src/main/resources/script/cbms_scriptsV4/V4.01.sql:
```diff
@@ -0,0 +1,73 @@
+SET FOREIGN_KEY_CHECKS=1;
+
+/*
+Adding one asset type for engin with dummy values
```
engin -> engine

JayChouteau commented on src/main/resources/script/cbms_scriptsV4/V4.01.sql:
```diff
@@ -0,0 +1,73 @@
+
+-- -----------------------------------------------------
+-- alter Table `cbms`.`asset`
+-- removing type colomn, addning asset_type_id colomn and adding the association with Asset_type table
```
colomn -> column

JayChouteau commented on src/main/resources/script/cbms_scriptsV4/V4.01.sql:
```diff
@@ -0,0 +1,73 @@
+-- -----------------------------------------------------
+-- Table `cbms`.`asset_type`
+-- -----------------------------------------------------
+CREATE TABLE IF NOT EXISTS `cbms`.`asset_type` (
```
Implementation well done with the primary and foreign keys

JayChouteau commented on src/main/resources/script/cbms_scriptsV4/V4.01.sql:
```diff
@@ -0,0 +1,73 @@
+/**
+
+this script will add the asset_type table and make the association with asset
+the script will also add a value in the type and set all current asset's types to this new one
```
The script will also add an entry for this new asset type table and set all current asset's types (in the asset table) to this new asset type entry

JayChouteau commented on src/main/resources/script/cbms_scriptsV4/V4.01.sql:
```diff
@@ -0,0 +1,73 @@
+the script will also add a value in the type and set all current asset's types to this new one
+
+this script will also create a table to store the trained models
+the table will take the modele id, the asset type id and the dataset id as primaries keys
```
Instead of explaining HOW we are doing the table, I think it would be more relevant to explain WHAT the table is doing, i.e.: what is a trained model for and what is it holding.

PaulMicu replied:
I thought it was pretty self-explanatory but I can add some more details

JayChouteau commented on src/main/resources/script/cbms_scriptsV4/V4.01.sql:
```diff
@@ -0,0 +1,73 @@
+
+this script will also create a table to store the trained models
+the table will take the modele id, the asset type id and the dataset id as primaries keys
+it also has a colomn for storing the path to the model once we encapsulate that in a future issue
```
colomn -> column

JayChouteau commented on src/main/resources/script/cbms_scriptsV4/V4.01.sql:
```diff
@@ -0,0 +1,73 @@
+ALTER TABLE asset
+  add CONSTRAINT `fk_asset_asset_type1`
+    FOREIGN KEY (`asset_type_id`)
+    REFERENCES `cbms`.`asset_type` (`asset_type_id`);
```
Adding an empty line after this one would help separate the logic blocks between asset alteration and trained model table dropping/creation

JayChouteau commented on src/main/resources/script/cbms_scriptsV4/V4.01.sql:
```diff
@@ -0,0 +1,73 @@
+  `model_id` INT(11) NOT NULL,
+  `asset_type_id` INT(11) NOT NULL,
+  `dataset_id` INT(11) NOT NULL,
+  `location` VARCHAR(255) NULL,
```
Since other entities, such as assets, also have an attribute called "location" that refers to a physical location, this name might lead to confusion. Would be more relevant to call it "path" or "path_location" or else.

JayChouteau reviewed (changes requested):
Logic and coding are structured, clear, and complete. There are some parts that will refinement depending on the information/direction the PO gives us, so adding a few TODOs would help us remember to check on him. I left some review changes mainly on comments, possible todos and on 1 naming issue. Very good work.

JayChouteau commented:
@PaulMicu 
The asset type table doesn't seem flexible in the sense that it doesn't allow to add more recommendations possibilities and values. This comment is opened to design discussion and was originally suggested by @rdsaliba (and I agreed with him). In our design, I feel like we should let the user determine the number of recommendations possible for a system type along with the name/display of the recommendation and the lower bound associated with it. 
-> An example of this design would be to have a table "status" relating to "asset_type". The "status" table would have asset_type_id as a foreign key reference, status_id, status name/display, and a lower boundary as attributes (columns). The asset_type would hold the asset_type_id and name attributes.

PaulMicu commented:
@JayChouteau I see what you mean. It's not a bad idea, it will make the changes to make it a little more versatile. I'll implement this new criteria asap

PaulMicu commented:
After the latest changes the database model looks like this : 
![V4 01](https://user-images.githubusercontent.com/35611221/99129758-76237480-25dc-11eb-93de-ab397e9e9cbb.png)

rdsaliba reviewed (approved):
Thank you for making the adjustments @PaulMicu. We'll see how this works out in the future.

PaulMicu committed:
Added new parameter table

Expanded usability by allowing the user to drive the number of parameters used for the recommendation

JayChouteau commented:
@PaulMicu Good job for proceeding with the requested changes. I think this new design makes it a lot easier to accept different kinds of input and nice modifications to the rest of the changes. We are good to go.
Small question:
- It seems we are merging into master, shouldn't we merge into develop?
